pageserver/src/pgdatadir_mapping.rs

```diff
@@ -887,7 +887,7 @@ impl<'a> DatadirModification<'a> {
         ctx: &RequestContext,
     ) -> Result<(), RelationError> {
         if rel.relnode == 0 {
-            return Err(RelationError::AlreadyExists);
+            return Err(RelationError::InvalidRelnode);
```

Comment:
```anyhow::ensure!(rel.relnode != 0, RelationError::InvalidRelnode);```
this is how all of the code in this file looks like

Reply:
Fixed

Reply:
@shanyp : here I want to return a `RelationError`
`anyhow::ensure!` returns an `anyhow::Error`

Reply:
I was initially thinking the anyhow::ensure should support this, but apparently not: https://docs.rs/anyhow/latest/anyhow/macro.ensure.html:

> The surrounding function’s or closure’s return value is required to be Result<_,[anyhow::Error](https://docs.rs/anyhow/latest/anyhow/struct.Error.html)>.

So it will always create `anyhow::Error` out of the passed in type (or so I hope, will confirm). But, this is fine to leave like this. What this file needs is to get rid of all rel.relnode checks via `std::num::NonZero*` types and only do this once, but there's probably some caveats which makes this transformation hard to write.